### ISSUE TITLE
Proxy texture returns incorrect property from VisualServer

### ISSUE THREAD

**Issue by buresu** [godotengine/godot#34987]
**Godot version:**
Current master ff173ff770c4498f4527676838ef8097917d0dda

**OS/device including version:**
All

**Issue description:**
Textures such as viewportTexture and ProxyTexture have a proxy texture ID and return it as their own ID.
https://github.com/godotengine/godot/blob/master/scene/main/viewport.cpp#L115-L119
https://github.com/godotengine/godot/blob/master/scene/resources/texture.cpp#L1932-L1935

If we get property from VirtualServer using them, the property (like width and height) is incorrect, because VisuaServer returns proxy texture property.
https://github.com/godotengine/godot/blob/master/drivers/gles3/rasterizer_storage_gles3.cpp#L1465-L1472

This problem can be solved by overriding the texture with RasterizerStorageGLES3::Texture::get_ptr(), but I'm not sure that is the right way.
https://github.com/godotengine/godot/blob/master/drivers/gles3/rasterizer_storage_gles3.h#L320-L326

**Minimal reproduction project:**
```
extends Spatial

func _ready():
	var image_texture = ImageTexture.new()
	image_texture.create(512, 512, Image.FORMAT_RGB8)
	var proxy_texture = ProxyTexture.new()
	proxy_texture.set_base(image_texture)
	print(VisualServer.texture_get_width(image_texture.get_rid())) # 512
	print(VisualServer.texture_get_width(proxy_texture.get_rid())) # 0
	pass 

```

**Comment by KoBeWi** [comment]
Still valid in 3.2.4 beta3

**Comment by UltraWelfare** [comment]
I think I just came across this issue on `ProxyTexture` from trying to fix #44529
> Textures such as viewportTexture and ProxyTexture have a proxy texture ID and return it as their own ID.

I was trying to get a `Ref<Image>` from `VisualServer::texture_get_data(RID)` by passing the `ProxyTexture::get_rid()` function but in the end it was showing no texture.

I think the aforementioned quote was causing the issue because it was requesting the image data from the proxy RID.

So I assume this line:
https://github.com/godotengine/godot/blob/a4d25a115ed65b05bb6e2b575c0d9ddef8a7a123/scene/resources/texture.cpp#L1932-L1935
should return `base->get_rid()` instead of `proxy`.

**Comment by buresu** [comment]
ProxyTexture has been removed in Godot 4 (PR #64595).
Therefore, I am closing this issue.
